connect/runtime/src/test/java/org/apache/kafka/connect/integration/ConnectorRestartApiIntegrationTest.java

```diff
@@ -0,0 +1,425 @@
+        connectBuilder = connectBuilderWithNumWorkers(ONE_WORKER); // true is the default, setting here as example
+        // get connector handles before starting test.
+        connectorHandle = RuntimeHandles.get().connectorHandle(CONNECTOR_NAME);
+    }
+
+    private EmbeddedConnectCluster.Builder connectBuilderWithNumWorkers(int numWorkers) {
+        return new EmbeddedConnectCluster.Builder()
+                .name("connect-cluster")
+                .numWorkers(numWorkers)
+                .workerProps(workerProps)
+                .brokerProps(brokerProps)
+                .maskExitProcedures(true);
+    }
+
+    @After
+    public void close() {
+        RuntimeHandles.get().deleteConnector(CONNECTOR_NAME);
+        // stop all Connect, Kafka and Zk threads.
+        connect.stop();
+    }
+
+    @Test
+    public void testRestartUnknownConnectorNoParams() throws Exception {
+        String connectorName = "Unknown";
+        connect = connectBuilder.build();
+        // start the clusters
+        connect.start();
+
+        // Call the Restart API
+        String restartEndpoint = connect.endpointForResource(
+                String.format("connectors/%s/restart", connectorName));
+        Response response = connect.requestPost(restartEndpoint, "", Collections.emptyMap());
+        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatus());
+
+    }
+
+    @Test
+    public void testRestartUnknownConnector() throws Exception {
+        restartUnknownConnector(false, false);
+        restartUnknownConnector(false, true);
+        restartUnknownConnector(true, false);
+        restartUnknownConnector(true, true);
+    }
+
+    private void restartUnknownConnector(boolean onlyFailed, boolean includeTasks) throws Exception {
+        String connectorName = "Unknown";
+        connect = connectBuilder.build();
+        // start the clusters
+        connect.start();
+
+        // Call the Restart API
+        String restartEndpoint = connect.endpointForResource(
+                String.format("connectors/%s/restart?onlyFailed=" + onlyFailed + "&includeTasks=" + includeTasks, connectorName));
+        Response response = connect.requestPost(restartEndpoint, "", Collections.emptyMap());
+        assertEquals(Response.Status.NOT_FOUND.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void testRestartOnlyConnector() throws Exception {
+        runningConnectorAndTasksRestart(false, false, 1, allTasksExpectedRestarts(0), false);
+    }
+
+    @Test
+    public void testRestartBoth() throws Exception {
+        runningConnectorAndTasksRestart(false, true, 1, allTasksExpectedRestarts(1), false);
+    }
+
+    @Test
+    public void testRestartNoopOnlyConnector() throws Exception {
+        runningConnectorAndTasksRestart(true, false, 0, allTasksExpectedRestarts(0), true);
+    }
+
+    @Test
+    public void testRestartNoopBoth() throws Exception {
```

Comment:
Again, "both" what?

Reply:
Fixed and changed all test names to be explicit